es.go

```diff
@@ -767,9 +767,28 @@ func (c *Client) ModifyAliases(actions []AliasAction) error {
 //
 // Use case: You want to remove an index and all of its data.
 func (c *Client) DeleteIndex(indexName string) error {
+	return c.DeleteIndexWithQueryParameters(indexName, nil)
+}
+
+// Delete an index in the cluster with query parameters.
+//
+// Use case: You want to remove an index and all of its data. You also want to
+// specify query parameters such as timeout.
+func (c *Client) DeleteIndexWithQueryParameters(indexName string, queryParamMap map[string][]string) error {
+	queryParams := make([]string, len(queryParamMap))
+	for key, value := range queryParamMap {
+		queryParams = append(queryParams, fmt.Sprintf("%s=%s", key,
+			strings.Join(value, ",")))
+	}
+	queryString := strings.Join(queryParams, "&")
+
+	var urlBuilder strings.Builder
+	urlBuilder.WriteString(fmt.Sprintf("%s?%s", indexName, queryString))
```

Comment:
just a nit: I guess we could have used `fmt.Sprintf("%s?%s", indexName, queryString)` instead of a `strings.Builder`? If we really need to use a Builder (for performance reasons) we can do:

```go
...
urlBuilder.WriteString(indexName)
urlBuilder.WriteString(queryString)
...
```

Reply:
This make the code a lot easier to read.  Thanks a lot for the feedback. I think readability > performance since this is a operation lib 🙂